File: Ontology/DataPropertyAxiom.cs
﻿namespace Ontology
{
    public abstract class DataPropertyAxiom:
        Axiom,
        IDataPropertyAxiom
    {
        private readonly IDataPropertyExpression _dataPropertyExpression;

        protected DataPropertyAxiom(
            IOntology               ontology,
            IDataPropertyExpression dataPropertyExpression
            ) : base(ontology)
        {
            _dataPropertyExpression = dataPropertyExpression;
        }

        IDataPropertyExpression IDataPropertyAxiom.DataPropertyExpression => _dataPropertyExpression;
    }
}

File: Web/ClientApp/src/app/Ontology/DataPropertyAxiom.ts
import { Axiom } from "./Axiom";
import { IDataPropertyAxiom } from "./IDataPropertyAxiom";
import { IOntology } from "./IOntology";

export class DataPropertyAxiom
    extends Axiom
    implements IDataPropertyAxiom
{
    constructor(
        ontology: IOntology
        )
    {
        super(ontology);
    }
}
